PreferClang.cmake

```diff
@@ -1,7 +1,7 @@
 # Copyright [2019] <Malinovsky Rodion> (rodionmalino@gmail.com)
 
 # Prefer Clang compiler
-if (NOT CMAKE_CXX_COMPILER AND NOT CMAKE_C_COMPILER AND NOT DEFINED ENV{CXX} AND NOT DEFINED ENV{CC} )
+if (NOT CMAKE_CXX_COMPILER AND NOT CMAKE_C_COMPILER AND NOT DEFINED ENV{CXX} AND NOT DEFINED ENV{CC} AND NOT APPLE)
     message(STATUS "Compiler is not set explicitly. Using clang++")
```

Comment:
Is it possible to have normal clang on mac? I think proper handling is to check whether clang is available rather that checking is it apple or not. What if platform is win?

Reply:
@malirod, clang++ is used by default on macOS so I thought that there is no need to respecify it.

Anyway - reverted back.